FlyAirDB/1 Table.sql

```diff
@@ -0,0 +1,46 @@
+use FlyAirDB
+go
+drop table if exists FlightInfo
+go 
+create table dbo.FlightInfo (
+    FlightInfoID int primary key not null identity,
+    FlightNum char(6) not null 
+        Constraint ck_FlightInfo_FlightNum_must_be_6_characters_start_with_FLY_and_contain_3_num_starting_from_001
+        check(FlightNum like 'FLY[0-9][0-9][0-9]' and FlightNum not like 'fly000' ), 
+    DepartureAirport char(3) not null constraint ck_FlightInfo_DepartureAirport_cannot_contain_spaces check(DepartureAirport not like '% %'), 
+    DepartureCountry varchar(25) not null constraint ck_FlightInfo_DepartureCountry_cannot_be_blank check(DepartureCountry > ''), 
+    ArrivalAirport char(3) not null constraint ck_FlightInfo_ArrivalAirport_cannot_contain_spaces check(ArrivalAirport not like '% %'), 
+    ArrivingCountry varchar(25) not null constraint ck_FlightInfo_ArrivingCountry_cannot_be_blank check(ArrivingCountry > ''), 
+    DepartingDateTime datetime not null constraint ck_FlightInfo_DepartingDateTime_cannot_be_more_then_a_year_from_today
+        check(DepartingDateTime < getdate() + 365), --It seems to work fine even though I didn't specify that I am adding days. 
+        --Should probably add constraint that it shouldnt be in the pass, but it would conflict with sample data.
+    ArrivingDateTime datetime not null, 
+    FirstName varchar(30) not null constraint ck_FlightInfo_FirstName_cannot_be_blank check(FirstName > ''), 
+    LastName varchar(30) not null constraint ck_FlightInfo_LastName_cannot_be_blank check(LastName > ''), 
+    DOB date not null constraint ck_FlightInfo_DOB_cannot_be_in_the_future check(DOB <= getdate()), 
+    PassAddress varchar(150) not null constraint ck_FlightInfo_Address_cannot_be_blank check(PassAddress > ''), 
+    CheckedIn bit not null default 0, 
+    PassportNum int null constraint ck_FlightInfo_PassportNum_must_be_9_num_and_not_start_with_0 check(PassportNum between 100000000 and 999999999),    
+    IssueDate date null,
+    ExpiryDate date null, 
+    Nationality varchar(30) null constraint ck_FlightInfo_Nationality_cannot_be_blank check(Nationality > ''),
+    constraint ck_FlightInfo_CheckedIn_can_only_occur_30_days_from_departure
+        check((datediff(day,getdate(),DepartingDateTime) > 30 and CheckedIn = 0) or datediff(day,getdate(),DepartingDateTime) < 30),
+    constraint ck_FlightInfo_ArrivalDateTime_must_be_after_depaurturedatetime_and_less_then_3_days_later
+        check(ArrivingDateTime between DepartingDateTime and DepartingDateTime + 3), 
+    constraint ck_FlightInfo_DOB_must_be_between_16_and_90_by_departure_date check(datediff(day,DOB,DepartingDateTime) between 5840 and 32850),
```

Comment:
I changed it, however if I put 89 it will block some people before they are 90, and if I put 90 it will allow some people that are already 90

Reply:
It's ok to allow 90. Except if you want to check on year, month and day. Which will be a long (but correct) way.

Reply:
I found a way which seems to work perfect `format(dateadd(day,DATEDIFF(day,DOB,getdate()),0),'yy')`

Reply:
I think that this code `check(format(dateadd(day,DATEDIFF(day,DOB,DepartingDateTime),0),'yy') < 90 and datediff(year,DOB,DepartingDateTime) >= 16)` will only allow under 90 and will allow anyone turning 16 that year. Please correct me if I am wrong.

Reply:
I added in `and datediff(year,DOB,DepartingDateTime) < 92` because I realized that the first part is only going to work until someone is 100, at that point it won't stop the record from being added.

Reply:
I was trying to figure out what you meant by checking month and day. Did you mean something like this: 
` check (datediff(year,DOB,DepartingDateTime) between 16 and 89 
or (datediff(year,DOB,DepartingDateTime) = 90 and month(DOB) > month(DepartingDateTime)
or (datediff(year,DOB,DepartingDateTime) = 90 and month(DOB) = month(DepartingDateTime) and day(DOB) >= day(DepartingDateTime)))`